### PR TITLE
Improve Alica Tests

### PR DESCRIPTION
**Pull request by abolinsky**
# MERGE https://github.com/rapyuta-robotics/alica/pull/17 FIRST
The step engine is used to ensure synchronized steps, code that takes the steps is simplified, and tests now fail an assert if a SIGSEGV or SIGABRT signal is raised, allowing us to see in which test the signal occurred to aid in the process of debugging.

### CODE REVIEW

File: alica_tests/src/test/test_alica_authority.cpp [hunk 1]
```diff
@@ -104,6 +105,8 @@ TEST(AllocationDifference, MessageCancelsUtil) {
 }
 
 TEST_F(AlicaEngineAuthorityManager, authority) {
+    ASSERT_EQ(setjmp(restore_point), 0);
```
Comment (hendrik-skubch) [inline alica_tests/src/test/test_alica_authority.cpp]: This exposes way too much internals.
For the case without fixture you can define a macro that calls `TEST`.

With fixture, it is a little more complicated. But at the very least you can provide a macro `ASSERT_NO_TRAP` (or think of a better name).

File: alica_tests/src/test/test_alica_behaviourtrigger.cpp [hunk 2]
```diff
@@ -109,15 +111,15 @@ TEST_F(AlicaBehaviourTrigger, triggerTest) {
     }
     alicaTests::TestWorldModel::getOne()->trigger1->run();
     alicaTests::TestWorldModel::getOne()->trigger2->run();
-    this_thread::sleep_for(duration);
+    step(ae);
```
Comment (hendrik-skubch) [inline alica_tests/src/test/test_alica_behaviourtrigger.cpp]: This is not the step you want.
BehaviourTrigger are independent of engine steps, they intentionally run asynchronously.

The Improve Query PR has some changes to this test.

File: alica_tests/src/test/test_alica_problem_composition.cpp [hunk 3]
```diff
@@ -76,15 +77,12 @@ class AlicaProblemCompositionTest : public ::testing::Test {
...
-    ae->stepNotify();
-    this_thread::sleep_for(chrono::milliseconds(33));
-    for (int i = 0; i < 5; ++i) {
-        while (!ae->getPlanBase()->isWaiting()) {
-            this_thread::sleep_for(chrono::milliseconds(33));
-        }
-        ae->stepNotify();
-    }
+
+    for (int i = 0; i < 5; ++i) { step(ae); }
```
Comment (hendrik-skubch) [inline alica_tests/src/test/test_alica_problem_composition.cpp]: i<6

File: alica_tests/src/test/test_alica_ros_proxy.cpp [hunk 4]
```diff
@@ -1,5 +1,6 @@
 #include <iostream>
 #include <typeinfo>
+#include <test_alica.h>
```
Comment (gauthamm) [inline alica_tests/src/test/test_alica_ros_proxy.cpp]: Doesn't look like its being used

File: alica_tests/src/test/test_alica_behaviourtrigger.cpp (hunk 2 above)
Comment (abolinsky) [reply alica_tests/src/test/test_alica_behaviourtrigger.cpp]: I'll allow the Query PR to be merged so I can incorporate those changes into here before addressing this.

**Comment by abolinsky** [comment]
@hendrik-skubch I've addressed your comments

**Pull request merged**